src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,85 @@
 package core.basesyntax;
 
+import java.util.LinkedList;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+
+    private LinkedList<Entry<K, V>>[] table;
+    private int size;
+
+    public MyHashMap() {
+        this(DEFAULT_CAPACITY);
+    }
+
+    public MyHashMap(int initialCapacity) {
+        table = new LinkedList[initialCapacity];
+        size = 0;
+    }
+
+    static class Entry<K, V> {
+        private K key;
+        private V value;
+
+        Entry(K key, V value) {
+            this.key = key;
+            this.value = value;
+        }
+    }
 
-    @Override
     public void put(K key, V value) {
+        if (size >= table.length * LOAD_FACTOR) {
+            resize();
+        }
+
+        int index = getIndex(key);
+        if (table[index] == null) {
+            table[index] = new LinkedList<>();
+        }
 
+        for (Entry<K, V> entry : table[index]) {
+            if ((key == null && entry.key == null) || (key != null && key.equals(entry.key))) {
+                entry.value = value;
+                return;
+            }
+        }
+
+        table[index].add(new Entry<>(key, value));
```

Comment:
Diamond operator. Use `new Entry<K, V>(key, value)` instead of `new Entry<>(key, value)` to explicitly specify the type parameters for clarity and consistency.